packages/ember-engines/addon/components/link-to-component.js

```diff
@@ -1,13 +1,21 @@
-import LinkComponent from '@ember/routing/link-component';
-import { getOwner } from '@ember/application';
+import RoutingLinkComponent from '@ember/routing/link-component';
+import Application, { getOwner } from '@ember/application';
 import { computed, get, set } from '@ember/object';
 import { typeOf } from '@ember/utils';
 import { assert, deprecate } from '@ember/debug';
-import { macroCondition, dependencySatisfies } from '@embroider/macros';
+import { macroCondition, getOwnConfig, importSync, dependencySatisfies } from '@embroider/macros';
 
 let LinkTo;
+let LinkComponent;
 
-if (macroCondition(dependencySatisfies('ember-source', '>= 3.24.1'))) {
+if (macroCondition(dependencySatisfies('@ember/legacy-built-in-components', '*'))) {
```

Comment:
@BobrImperator I assume we need `@ember/legacy-built-in-components` to be listed in `dependencies` otherwise this line always gonna return `false` in real world app.

As an example you may see #794

Reply:
You're right, though for now I assumed that users would need to install it themselves just like in scenarios, but that's not exactly the best idea.